### PR TITLE
Make all struct fields public and others @sdroege PR's

### PR DESCRIPTION
**Pull request by EPashkin**
Apply https://github.com/gtk-rs/gir/pull/341, https://github.com/gtk-rs/gir/pull/342, https://github.com/gtk-rs/gir/pull/343, https://github.com/gtk-rs/gir/pull/344, https://github.com/gtk-rs/gir/pull/345, https://github.com/gtk-rs/gir/pull/346

### CODE REVIEW

File: glib-sys/src/lib.rs [hunk 1]
```diff
@@ -1589,7 +1589,8 @@ bitflags! {
 // Unions
 pub type GDoubleIEEE754 = c_void; // union
 pub type GFloatIEEE754 = c_void; // union
-pub type GMutex = c_void; // union
+#[repr(C)]
+pub struct GMutex(u64); // union
```
Comment (sdroege) [inline glib-sys/src/lib.rs]: I just noticed that this is not 100% correct with the alignment on 32 bit architectures. New pull request in ~1 hour when I'm back home

Comment (sdroege) [inline glib-sys/src/lib.rs]: See https://github.com/gtk-rs/gir/pull/346

**Comment by sdroege** [comment]
Looks good to me, thanks

**Comment by EPashkin** [comment]
@GuillaumeGomez, take look at this please

**Comment by GuillaumeGomez** [comment]
As long as it's only in sys structs, we can put every fields public, it's fine. Thanks @EPashkin!

**Comment by sdroege** [comment]
🎉

**Pull request merged**